analysis_driver/quality_control/calculate_relatedness.py

```diff
@@ -11,6 +14,111 @@ class RelatednessStage(segmentation.Stage):
     def gatk_outfile(self):
         return os.path.join(self.job_dir, self.dataset.name + '_genotype_gvcfs.vcf')
 
+    def family_id(self, sample_id):
+        family_id = clarity.get_sample(sample_id).udf.get('Family ID')
+        if not family_id:
+            return 'No_ID'
+        return family_id
+
+    def relationship(self, member):
+        relationship = clarity.get_sample(member).udf.get('Relationship')
+        if not relationship:
+            return 'Other'
+        return relationship
+
+    def sex(self, member):
+        sex = clarity.get_sample(member).udf.get('Sex')
+        if not sex:
+            return 'No_Sex'
+        return sex
+
+
+class ParseRelatedness(RelatednessStage):
+    parse_method = Parameter()
+
+    @property
+    def reformat_relatedness_file(self):
+        relatedness_reformat = (os.path.join(self.job_dir, self.dataset.name + '.relatedness2_reformat'))
+        with open(os.path.join(self.job_dir, self.dataset.name + '.relatedness2')) as openfile:
+            readfile = openfile.read()
+            readfile_csv = re.sub('\t', ',', readfile)
+            with open(relatedness_reformat, 'w') as outfile:
+                outfile.write(readfile_csv)
+        return relatedness_reformat
+
+    @property
+    def peddy_file(self):
+        return os.path.join(self.job_dir, self.dataset.name + '.ped_check.csv')
+
+    def write_all_fields(self, values):
+        with open(os.path.join(self.job_dir, self.dataset.name + '.relatedness_output'), 'w') as outfile:
+            for r in values:
+                sample1 = r[0]
+                sample2 = r[1]
+                relatedness_values = r[2:]
+                line = [sample1,
+                        self.family_id(sample1),
+                        self.relationship(sample1),
+                        sample2,
+                        self.family_id(sample2),
+                        self.relationship(sample2)]
+                line.extend(relatedness_values)
+                outfile.write('\t'.join(line) + '\n')
+
+    def get_columns(self, filename, column_headers):
+        columns_to_return = []
+        with open(filename) as openfile:
+            csvfile = csv.DictReader(openfile)
+            for line in csvfile:
+                columns_to_return.append([line[i] for i in column_headers])
+        return columns_to_return
+
+    def parse_all(self):
+        exit_status = 0
+        try:
+            peddy_relatedness = self.get_columns(self.peddy_file, ['sample_a', 'sample_b', 'rel'])
+            peddy_vcftools_relatedness = []
+            with open(self.reformat_relatedness_file) as openfile:
+                csvfile = csv.DictReader(openfile)
+                for line in csvfile:
+                    relatedness_samples = (Counter([line['INDV1'], line['INDV2']]))
+                    [peddy_vcftools_relatedness.append([i[0],
```

Comment:
List comprehension for calling `list.append`? A simple `for` loop would be easier to read